api/javatests/org/openyolo/api/CredentialClientTest.java

```diff
@@ -282,17 +298,57 @@ public void getSaveIntent_threeKnownProviders() throws Exception {
     @SuppressWarnings("WrongConstant")
     @Test
     public void getSaveIntent_noKnownProviders() throws Exception {
-        addUnknownProviders("com.example.superpassword", "com.example.locker");
+        addUnknownProviders(UNKNOWN_PROVIDER_1, UNKNOWN_PROVIDER_2);
 
         // With no known providers, the preferred provider heuristic will determine there is no
         // preferred provider. therefore, a provider picker should be displayed.
-        Intent saveIntent = credentialClient.getSaveIntent(testCredential);
+        final CredentialSaveRequest request = CredentialSaveRequest.fromCredential(testCredential);
+        final Intent saveIntent = credentialClient.getSaveIntent(request);
+
         assertThat(saveIntent).isNotNull();
         assertThat(saveIntent.getComponent().getClassName())
                 .endsWith("ProviderPickerActivity");
     }
 
     @Test
+    public void getCredentialSaveResult_withNullIntent_returnsDefaultResult() {
+        final CredentialSaveResult result =
+                credentialClient.getCredentialSaveResult(null /* intent*/);
+
+        assertThat(result.getResultCode()).isEqualTo(CredentialSaveResult.CODE_UNKNOWN);
+        assert(result.getAdditionalProperties()).isEmpty();
```

Comment:
assert -> assertThat - I'm actually not sure what the semantics of this is, given that assert is a keyword.